automatic/youtube-dl/update.ps1

```diff
@@ -11,14 +8,14 @@ function global:au_BeforeUpdate {
     New-Item -ItemType Directory "$PSScriptRoot\tools"
   }
   $Latest.FileName = Get-WebFileName $Latest.URL32 'youtube-dl.exe'
-  $filePath = "$PSScriptRoot\tools\$($Latest.FileName)"
-  Get-WebFile $Latest.URL32 $filePath
+  $FilePath = "$PSScriptRoot\tools\$($Latest.FileName)"
+  Get-WebFile $Latest.URL32 $FilePath
```

Comment:
Is there a reason to PascalCase the variable names (I don't see them as function parameters)? They should be camelCase.

Reply:
Do we have guidance that suggests that? Our code, I think, doesn't reflect it.

In the original version of this file, we seem to have at least three different types of variable casing 😅:
- `$Latest`
- `$domain`
- `$download_page`
- `$filename` (half point, arguable)

I know I spoke about a coding standards document for PowerShell code way back when - I don't think it ever happened. I've always been a fan of / used by default the [PoshCode PowerShell Practice and Style guide](https://poshcode.gitbook.io/powershell-practice-and-style/style-guide/code-layout-and-formatting) which says:

> PowerShell uses PascalCase for all public identifiers: module names, function or cmdlet names, class, enum, and attribute names, public fields or properties, global variables and constants, etc. [...]
> If you wish, you may use camelCase for variables within your functions (or modules) to distinguish private variables from parameters, but this is a matter of taste.

C# certainly uses camel case for private variables more frequently. To me, the above indicates that it's the second-choice in PowerShell.

Reply:
There is lots in that guide that criticises itself and then trips up over itself then criticises itself again. It doesn't ever make statements, just guidelines, recommendations and personal opinions. I really wish it was more definitive. Using it however, it does suggest:

> If you wish, you may use camelCase for variables within your functions (or modules) to distinguish private variables from parameters, but this is a matter of taste. Shared variables should be distinguished by using their scope name, such as $Script:PSBoundParameters or $Global:DebugPreference. If you are using camelCase for a variable that starts with a two-letter acronym (where both letters are capitalized), both letters should be set to lowercase (such as adComputer).

`$Latest` is a global variable, so using the style guide again:

> PowerShell uses PascalCase for all public identifiers: module names, function or cmdlet names, class, enum, and attribute names, public fields or properties, global variables and constants, etc.

`$domain` and `$filename` are camelCase (there just is no second word to Case). 

If the change was made for consistency then I'd agree. If you changed all of the variables to PascalCase then I could see your reasoning. but I don't see the need for this change at all as all it's doing is renaming a variable from camelCase to PascalCase and nothing more.

Reply:
> [...] $filename are camelCase (there just is no second word to Case)

Impossible to tell really, but yes, I gave it a half point. :) File name and user name always get me as fun compound words that I'm never sure if I prefer split and (arguably) work either way as a definition. Yay, naming/words.

If we were more consistent, had written guidelines, or relevant formatter rules in place this would be easier! :) We've both quoted "it's a matter of taste" to each other, now, and the chunk about public identifiers - though I had them the other way around.

Of course, the downside of the many-scoped, joyfully fun layout of AU would likely mean that any automated rule would completely fail to parse the difference between global variables (that haven't been prefixed with `$global:`, which is, in my opinion, the better way to show scope) and local variables, but... there's only so much fun we can have, I suppose.

No matter. Sorted. Or resorted, or unsorted, or something. :)